contracts/strategy/LiquidationStrategy.sol

```diff
@@ -144,80 +162,151 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
+        if (_poolFee == 0) revert ZeroAmount();
+        if (keccak256(abi.encode(_permit2, _universalRouter, _poolFee)) == keccak256(abi.encode(uniswapV3Info[_universalRouter].permit2, _universalRouter, uniswapV3Info[_universalRouter].poolFee))) revert SameV3Info();
+        uniswapV3Info[_universalRouter] = UniswapV3Info({ permit2: _permit2, poolFee: _poolFee });
+        emit LogSetV3Info(_permit2, _universalRouter);
+    }
 
-        WXDC.safeTransfer(strategyManager, _amount);
-        emit LogShutdownWithdrawWXDC(strategyManager, _amount);
+    function setUSDTInPath(address _usdToken) external onlyStrategyManager {
+        if (_usdToken == address(0)) revert ZeroAddress();
+        usdToken = ERC20(_usdToken);
+        emit LogSetUSDTInPath(_usdToken);
     }
 
-    /// @notice Allows the strategy manager to sell WXDC held by the contract.
+    /// @notice Allows the strategy manager to sell Collateral, held by the contract, to UniV2.
     /// @dev Only the current strategy manager can call this function.
-    /// This allows for managing the WXDC obtained through liquidations.
+    /// This allows for managing the Collateral obtained through liquidations.
     /// @param _router The UniswapV2Router02 (or a fork) used for the sale.
-    /// @param _amount The amount of WXDC to sell.
+    /// @param _amount The amount of Collateral to sell.
     /// @param _minAmountOut The minimum amount of FXD to accept for the sale.
-    function sellWXDC(IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
-        require(address(_router) != address(0), "LiquidationStrategy: zero address");
-        require(_amount > 0, "LiquidationStrategy: zero amount");
-        require(_amount <= idleWXDC.WXDCAmount, "LiquidationStrategy: wrong amount");
+    function sellCollateralV2(address _collateral, IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
+        if (address(_router) == address(0)) revert ZeroAddress();
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].collateralAmount) revert WrongAmount();
+
+        // Calculate the cost of Collateral in terms of FXD using the average price
+        uint256 _averagePriceOfCollateral = idleCollateral[_collateral].averagePriceOfCollateral;
+        uint256 costOfCollateralInFXD = _amount.mul(_averagePriceOfCollateral).div(WAD); 
+        uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+        
+        // Existing logic for selling Collateral
+        idleCollateral[_collateral].collateralAmount -= _amount;
+
+        _nullifyIdleCollateral(_collateral);
+
+        //if idleCollateral is nullified in previous line, no need to _adjustAmountNeededToPayDebt
+        if (idleCollateral[_collateral].collateralAmount != 0) {
+            _adjustAmountNeededToPayDebt(_collateral, _amount.mul(_averagePriceOfCollateral).div(WAD));
+        }
```

Comment:
so the execution flow looks like this right now 
```
 if (_amount > idleCollateral[_collateral].collateralAmount) revert WrongAmount();
...
idleCollateral[_collateral].collateralAmount -= _amount;
//  _nullifyIdleCollateral(_collateral); 
if (idleCollateral[_collateral].collateralAmount == 0) {
    delete idleCollateral[_collateral];
}
if (idleCollateral[_collateral].collateralAmount != 0) {
    _adjustAmountNeededToPayDebt(_collateral, _amount.mul(_averagePriceOfCollateral).div(WAD));
}
```
this still looks inefficient, we can make something like this
```
 if (_amount > idleCollateral[_collateral].collateralAmount) revert WrongAmount();
...
uint256 existingCollateral = idleCollateral[_collateral].collateralAmount;
if (existingCollateral == amount) {
    delete idleCollateral[_collateral];
} else {
    _adjustAmountNeededToPayDebt(_collateral, _amount.mul(_averagePriceOfCollateral).div(WAD));
}
```